src/ebay-textbox/textbox.tsx

```diff
@@ -56,34 +81,63 @@ const EbayTextbox: FC<EbayTextboxProps> = ({
         disabled: rest.disabled,
         label: floatingLabel,
         inputSize,
-        inputValue: controlledValue || value,
+        inputValue: controlledValue || inputValue,
         placeholder,
-        invalid
+        invalid,
+        onMount: onFloatingLabelInit
     })
 
     const handleFocus = (event?: FocusEvent<HTMLInputElement & HTMLTextAreaElement>) => {
-        onFocus(event)
+        onFocus(event, { value: event?.target?.value || defaultValue })
         onFloatingLabelFocus()
     }
 
     const handleBlur = (event: FocusEvent<HTMLInputElement & HTMLTextAreaElement>) => {
-        onBlur(event)
+        const newValue = event.target?.value
+        onBlur(event, { value: newValue })
+        if (newValue !== value) {
+            onChange(event, { value: newValue })
+            setValue(newValue)
+        }
         onFloatingLabelBlur()
     }
 
+    const handleKeyPress = (event?: KeyboardEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onKeyPress(event, { value: textbox?.value })
+    }
+    const handleKeyUp = (event?: KeyboardEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onKeyUp(event, { value: textbox?.value })
+    }
+    const handleKeyDown = (event?: KeyboardEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onKeyDown(event, { value: textbox?.value })
+    }
+
+    const handleInvalid = (event?: SyntheticEvent<HTMLInputElement & HTMLTextAreaElement>) => {
+        const textbox = event.target as any
+        onInvalid(event, { value: textbox?.value })
+    }
+
+    const handleButtonClick = (event?: KeyboardEvent<HTMLInputElement> & MouseEvent<HTMLInputElement>) => {
+        onButtonClick(event, { value })
```

Comment:
I don't think is necessary to pass the `value` here. Also, I am not sure the event is a HTMLInputElement since it is a IconButton or Icon, not sure how TS is not complaining here